src/plugins/content_management/public/content_client/content_client.tsx

```diff
@@ -21,19 +23,50 @@ export const queryKeyBuilder = {
   },
 };
 
+const addVersion = <I extends { contentTypeId: string; version?: Version }>(
+  input: I,
+  contentTypeRegistry: ContentTypeRegistry
+): I & { version: Version } => {
+  const contentType = contentTypeRegistry.get(input.contentTypeId);
+
+  if (!contentType) {
+    throw new Error(`Unknown content type [${input.contentTypeId}]`);
+  }
+
+  const version = input.version ?? contentType.version.latest;
+
+  const versionNumber = validateVersion(version);
+
+  if (versionNumber > parseInt(contentType.version.latest.substring(1), 10)) {
+    throw new Error(
+      `Invalid version [${version}]. Latest version is [${contentType.version.latest}]`
+    );
+  }
+
+  return {
+    ...input,
+    version,
+  };
+};
+
 const createQueryOptionBuilder = ({
   crudClientProvider,
+  contentTypeRegistry,
 }: {
   crudClientProvider: (contentType: string) => CrudClient;
+  contentTypeRegistry: ContentTypeRegistry;
 }) => {
   return {
-    get: <I extends GetIn = GetIn, O = unknown>(input: I) => {
+    get: <I extends GetIn = GetIn, O = unknown>(_input: I) => {
+      const input = addVersion(_input, contentTypeRegistry);
```

Comment:
thinking if we can do it in a central place. maybe `sendMessage` or the rpc_client? 

I assume you didn't do it there because then the other "custom" client-side clients won't get this automatically?

Reply:
Yes indeed. And in `sendMessage` it would not keep the RPC service "generic". It would know about CM stuff.